testapi.pm

```diff
@@ -2189,12 +2189,8 @@ sub upload_asset ($file, $public = undef, $nocheck = undef) {
     $cmd .= show_curl_progress_meter();
     my $basename = basename($file);
     $cmd .= autoinst_url("/upload_asset/$basename");
-    if ($nocheck) {
-        type_string("$cmd\n");
-    }
-    else {
-        return assert_script_run($cmd);
-    }
+    return assert_script_run($cmd) unless $nocheck;
+    type_string("$cmd\n");
```

Comment:
Please get #2553 done first as that will add the test coverage for this

Reply:
The idea was that with this change adding test coverage is easier. That's what we discussed today in the daily

Reply:
Why not just add it to the other branch? Making more untested changes seems counter-intuitive to me

Reply:
Well, either we merge here first and @asdil12 rebases or they pick my commit into their branch